tests/lookup/tests.py

```diff
@@ -1527,3 +1530,19 @@ def test_conditional_expression(self):
                 {"year": 2042, "century": "other"},
             ],
         )
+
+    def test_multivalued_join_reuse(self):
+        self.assertEqual(
+            Season.objects.get(Exact(F("games__home"), "NY"), games__away="Boston"),
+            self.s1,
+        )
+        self.assertEqual(
+            Season.objects.get(Exact(F("games__home"), "NY") & Q(games__away="Boston")),
+            self.s1,
+        )
+        self.assertEqual(
+            Season.objects.get(
+                Exact(F("games__home"), "NY") & Exact(F("games__away"), "Boston")
+            ),
+            self.s1,
+        )
```

Comment:
One thing to note here is that a `LEFT OUTER JOIN` will be generated instead of an `INNER JOIN` because the expression is too _opaque_ for the filter building logic to know that it can use an inner join here. This is more of an optimization problem that a bug though.

Reply:
Pushed a second commit that addresses this case.